Update Makefile

schmarta opened the pull request:
Add CFLAGS and default make target

schmarta committed:
Update Makefile

Add CFLAGS and default make target

CrazyWillBear reviewed (changes requested):
What does this do differently/improve? (not requesting a change, however it still must be addressed).

schmarta commented:
I added a default target to the Makefile, so when running `make` it will default to build. Then I spaced out the definitions at the top for ease of reading. I added CFLAGS which is a variable that is utilized later for compiler flags, it contains `-march=native -O2` by default. -march defines the CPU optimizations to include by default, this is set to native to use the builders CPU generation, for my CPU this would be Intel CoffeeLake, by using CPU optimizations you can gain faster speed when running your program (this is why gentoo becomes faster than normal Linux distros.) Finally, I added `-O2`, `-O` is the preset optimization flag, It simply enables multiple other optimization flags based on the level of optimization, https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html read here for more info. Last of all, I made the output binary have the name wilpac, so you can address the binary with `./bin/wilpac` instead of `./bin/main`.

CrazyWillBear commented:
Thanks for explaining, these changes make sense and I see the value they hold. Accepted!
